inst/htmljs/animint.js

```diff
@@ -2019,13 +2030,91 @@ var animint = function (to_select, json_file) {
       d3.select("title").text(response.title);
     }
     // Add plots.
+  
+  
+  var check_svg = function (row_number,col_number) {
+    var id="#row"+row_number+"col"+col_number
+    var parentElement= d3.select(id)
+    if (!parentElement.select('svg').empty()) {
+      return true
+    }else{
+      return false
+    }
+  }
+  
+  
+  var maximum_row= 0
+  var maximum_column= 0
+  var construct_outer_table = function () {
+  var count=0
+  
+  var find_maximum_dimensions = function () {
+  
+  
     for (var p_name in response.plots) {
-      add_plot(p_name, response.plots[p_name]);
+      count=count+1
+    if (response.plots[p_name].row>maximum_row){
+      maximum_row=response.plots[p_name].row
+    }
+    if (response.plots[p_name].col>maximum_column){
+      maximum_column=response.plots[p_name].col
+    }
+  }
+   }
+  find_maximum_dimensions()
+    var count_dimensions = Math.ceil(Math.sqrt(count))-1;
+    var outer_table = element.append("table").attr("id", "outerTable").style("width", "1500px").style("border", "1px solid white");
+    maximum_row=Math.max(maximum_row,count_dimensions)
+    maximum_column=Math.max(maximum_column,count_dimensions)
```

Comment:
yes, it would create a 3*10 grid when 1*10 grid is sufficient. it would not cause an issue. the rest of the positions in the grid would be just empty and of background color. if we want to make a grid which is not larger than required like in this case we could. instead of taking square root of count we can take square root of no_dedicated_location_count. in your example if would be 0. so var count_dimensions would be 0 too and maximum_row would then be then 1(or 0 if we want 0 index html id for each cell). please tell I my new logic is ignoring something.

Reply:
what is no_dedicated_location_count?
If we want to throw an error when there is inconsistency in the row/col params across different plots(some have, others do not). Can we just take the max of row/col value among all the plots if user has specified all of them?
When user does not specify any row/col position, use the logic of sqrt.

Reply:
no_dedicated_location_count is just a count of plots with no row and column parameter passed. should I revert the functionality that allows user to pass row and column in some plots and not in others ? and just give error if user is inconsistent ? Toby told to display warning/error. if we want to display  warning then we will need to keep my current functionality where it handles inconsistency

Reply:
is there a compelling use case for specifying row/col for some plots and not others?
I would always specify row/col for all plots, or none at all.

Reply:
if the user wants to put only certain plots together among the many plots then giving the flexibility to pass row and column in some plots and not in the others might be useful. even if the user does not pass row and column in any of the plots, they still go into the outer table which I create, so consistency despite browser interactivity is still maintained.